### PR TITLE
Feature379658: Support <inheritdoc /> with cref attribute

### PR DESCRIPTION
**Pull request by luyajun0205**
1. [AB#391893](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/391893) Support <inheritdoc /> with cref attribute
2. [AB#391894](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/391894) Support Property inheritdoc
3. [AB#391921](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/391921) Support partial inheritance for <inheritdoc />
3. Support cross repo inheritdoc
4. Add IPerson interface and Student class into test project to cover inheritdoc

### CODE REVIEW

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 1]
```diff
@@ -1398,83 +1423,133 @@ private void SetInheritDocForType(Type t)
...
+                        if (SetInheritDoc(inheritFrom.Docs, m.Docs))
                         {
-                            if (MembersByUid.ContainsKey(uid))
+                            InheritDocItemsByUid.Add(m.Uid, inheritFrom.Uid);
+                        }
+                    }
+                    else
+                    {
+                        m.CrossInheritdocUid = crefUid;
+                        Console.WriteLine($"no found cref uid:{crefUid} for inheritdoc on uid:{m.Uid}-->");
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: we don't need to write to console for this. We can comment out this line for now.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 2]
```diff
@@ -1398,83 +1423,133 @@ private void SetInheritDocForType(Type t)
...
             bool isInherit = false;
 
-            if (!string.IsNullOrEmpty(inheritFrom?.Summary))
+            if (!string.IsNullOrEmpty(inheritFrom?.Summary) && string.IsNullOrEmpty(inheritTo?.Summary))
             {
                 inheritTo.Summary = inheritFrom.Summary;
                 isInherit = true;
             }
-            if (inheritFrom?.Parameters?.Count > 0)
+            if (inheritFrom?.Parameters?.Count > 0 && inheritFrom?.Parameters?.Count == 0)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: will it be an empty array or just null?

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 3]
```diff
@@ -1398,83 +1423,133 @@ private void SetInheritDocForType(Type t)
...
                 isInherit = true;
             }
-            if (!string.IsNullOrEmpty(inheritFrom?.Returns))
+            if (!string.IsNullOrEmpty(inheritFrom?.Returns) && string.IsNullOrEmpty(inheritTo?.Returns))
             {
                 inheritTo.Returns = inheritFrom.Returns;
                 isInherit = true;
             }
-            if (!string.IsNullOrEmpty(inheritFrom?.Value))
+            if (!string.IsNullOrEmpty(inheritFrom?.Value) && string.IsNullOrEmpty(inheritTo?.Returns))
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: Value?

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs (hunk 1 above)
Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: updated.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs (hunk 2 above)
Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: updated.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs (hunk 3 above)
Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: updated.

**Review by TianqiZhang — APPROVED** [review]

**Pull request merged**